components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -71,13 +71,54 @@ open class LoginsStorage {
         return raw
     }
 
-    private func doOpen(_ key: String) throws {
+    /// Unlock the database and reads the salt.
+    ///
+    /// Throws `LockError.mismatched` if the database is already unlocked.
+    ///
+    /// Throws a `LoginStoreError.InvalidKey` if the key is incorrect, or if dbPath does not point
+    /// to a database, (may also throw `LoginStoreError.Unspecified` or `.Panic`).
+    open func getDbSaltForKey(key: String) throws -> String {
+        try queue.sync {
+            if self.raw != 0 {
+                throw LockError.mismatched
+            }
+            let ptr = try LoginsStoreError.unwrap { err in
+                sync15_passwords_get_db_salt(self.dbPath, key, err)
+            }
+            return String(freeingRustString: ptr)
+        }
+    }
+
+    /// Migrate an existing database to a sqlcipher plaintext header.
+    ///
+    /// Throws `LockError.mismatched` if the database is already unlocked.
+    ///
+    /// Throws a `LoginStoreError.InvalidKey` if the key is incorrect, or if dbPath does not point
+    /// to a database, (may also throw `LoginStoreError.Unspecified` or `.Panic`).
```

Comment:
I think it would be worth adding a comment here to warn that, if the application calls this without first reading and storing the salt from the existing db, then it will render the database unusable.